### PR TITLE
feat: add saber-plugin-search

### PR DESCRIPTION
**Pull request by krmax44**
**Summary**

`saber-plugin-local-search` adds a hyper-fast, easy to integrate and highly customizable search to your app. See [README](https://github.com/krmax44/saber/blob/4d3d9ec995bc2cbdddb74e62cf1b7eb0fcb50a2e/packages/saber-plugin-local-search/README.md) for more.

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] Feature
- [ ] Code style update
- [ ] Refactor
- [x] Docs
- [ ] Build-related changes
- [ ] Other, please describe:

If changing the UI/CSS related code, please provide the **before/after** screenshot:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)

You have tested in the following browsers: (Providing a detailed version will be better.)

- [ ] Chrome
- [x] Firefox
- [ ] Safari
- [ ] Edge
- [ ] IE

If adding a **new feature**, the PR's description includes:

- [ ] A convincing reason for adding this feature
- [x] Related documents have been updated
- [ ] Related tests have been updated

To avoid wasting your time, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

Original issue: https://github.com/saberland/saber-theme-tailsaw/issues/5

Published PR with canary version: <details><summary>Canary Versions</summary>
 - `extract-sfc-blocks@0.0.3-canary.366.524f251.0`
 - `create-blog@0.2.8-canary.366.524f251.0`
 - `create-site@0.3.2-canary.366.524f251.0`
 - `saber-highlight-css@0.1.1-canary.366.524f251.0`
 - `saber-highlighter-prism@0.3.2-canary.366.524f251.0`
 - `saber-log@0.2.1-canary.366.524f251.0`
 - `saber-markdown@0.1.1-canary.366.524f251.0`
 - `saber-plugin-feed@0.4.1-canary.366.524f251.0`
 - `saber-plugin-git-modification-time@0.1.1-canary.366.524f251.0`
 - `saber-plugin-google-analytics@0.1.1-canary.366.524f251.0`
 - `saber-plugin-image@0.1.1-canary.366.524f251.0`
 - `saber-plugin-meta-redirect@0.1.1-canary.366.524f251.0`
 - `saber-plugin-netlify-redirect@0.1.1-canary.366.524f251.0`
 - `saber-plugin-prismjs@0.2.2-canary.366.524f251.0`
 - `saber-plugin-pwa@0.4.1-canary.366.524f251.0`
 - `saber-plugin-query-posts@0.4.1-canary.366.524f251.0`
 - `saber-plugin-search@0.0.2-canary.366.524f251.0`
 - `saber-plugin-transformer-html@0.1.1-canary.366.524f251.0`
 - `saber-plugin-transformer-pug@0.1.1-canary.366.524f251.0`
 - `saber-utils@0.2.1-canary.366.524f251.0`
 - `saber@0.7.1-canary.366.524f251.0`</details>

### CODE REVIEW

**Comment by krmax44** [comment]
Here's a demo where you can see it in action: https://deploy-preview-31--saber-theme-tailsaw.netlify.com/

File: packages/saber-plugin-local-search/lib/saber-browser.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+export default ({ Vue }) => {
+  Vue.prototype.$searchPages = debounce(performSearch, 50, {
+    leading: true,
+    trailing: true
+  })
+}
+
+async function performSearch(query) {
+  if (!searchData) {
+    let publicUrl = this.$siteConfig.publicUrl || '/'
```
Comment (egoist) [inline packages/saber-plugin-local-search/lib/saber-browser.js]: Use `__PUBLIC_URL__` which defaults to `/`

Comment (egoist) [inline packages/saber-plugin-local-search/lib/saber-browser.js]:
```suggestion
    let publicUrl = __PUBLIC_URL__
```

File: packages/saber-plugin-local-search/lib/saber-browser.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    publicUrl += publicUrl.endsWith('/') ? '' : '/'
+    const locale = this.$localePath === '/' ? 'default' : this.$localePath
+    const request = await fetch(`${publicUrl}_saber/search/${locale}.json`) // eslint-disable-line no-undef
+    const data = await request.json()
```
Comment (egoist) [inline packages/saber-plugin-local-search/lib/saber-browser.js]: let's make it shorter, `const data = await fetch().then(res => res.json())`

**Comment by egoist** [comment]
what about adding an option to allow more search strategies? e.g. `adapter: 'local' | 'algolia' | ImplementYourOwn`.

**Comment by krmax44** [comment]
I unfortunately can't fully test it at the moment, but I'll try to do it when I'm back home in a couple of days. If someone of you wants to experiment in the meantime, feel free to do so :)

**Comment by krmax44** [comment]
Updated preview with Tailsaw: https://5d5e79095fbf530008ae7f60--saber-theme-tailsaw.netlify.com/

**Comment by krmax44** [comment]
Wait it's already in master? :D

**Comment by krmax44** [comment]
It's in master but the PR is not marked as merged :thinking:

**Comment by egoist** [comment]
> @krmax44: It's in master but the PR is not marked as merged 🤔

Yes GitHub is screwed 😅

image

**Comment by krmax44** [comment]
Alright, lol. I'll wait on the next package publish so the canary flag gets removed, then I'll merge in the Tailsaw search PR.

**Comment by egoist** [comment]
Published `saber-plugin-search@0.1.0`

**Pull request merged**